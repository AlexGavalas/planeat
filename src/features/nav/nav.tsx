import { Anchor, Group, Text } from '@mantine/core';
import { type KeyPrefix } from 'i18next';
import { useTranslation } from 'next-i18next';
import Link from 'next/link';
import { useRouter } from 'next/router';

type LinkItem = {
    href: string;
    label: KeyPrefix<'common'>;
};

const LINKS = [
    { href: '/home', label: 'home' },
    { href: '/meal-plan', label: 'view_weekly_meal' },
    { href: '/connections', label: 'connections.title' },
] as const satisfies readonly LinkItem[];

export const Nav = () => {
    const { t } = useTranslation();
    const { pathname } = useRouter();

    return (
        <Group>
            {LINKS.map(({ href, label }) => {
                const isCurrent = pathname === href;

                return (
                    <Link key={href} href={href}>
                        <Anchor component="span">
                            <Text td={isCurrent ? 'underline' : undefined}>
                                {t(label)}
                            </Text>
                        </Anchor>
                    </Link>
                );
            })}
        </Group>
    );
};
