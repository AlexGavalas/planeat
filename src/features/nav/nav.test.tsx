import { type NextRouter } from 'next/router';

import { renderWithUser } from '~test/utils';

import { Nav } from './nav';

jest.mock('next/router', () =>
    jest.requireActual<NextRouter>('next-router-mock'),
);

describe('<Nav />', () => {
    it('renders', () => {
        const { container } = renderWithUser(<Nav />);

        expect(container).toMatchSnapshot();
    });
});
