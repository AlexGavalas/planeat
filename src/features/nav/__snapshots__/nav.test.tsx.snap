// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Nav /> renders 1`] = `
<div>
  <style
    data-mantine-styles="classes"
  >
    @media (max-width: 35.99375em) {.mantine-visible-from-xs {display: none !important;}}@media (min-width: 36em) {.mantine-hidden-from-xs {display: none !important;}}@media (max-width: 47.99375em) {.mantine-visible-from-sm {display: none !important;}}@media (min-width: 48em) {.mantine-hidden-from-sm {display: none !important;}}@media (max-width: 61.99375em) {.mantine-visible-from-md {display: none !important;}}@media (min-width: 62em) {.mantine-hidden-from-md {display: none !important;}}@media (max-width: 74.99375em) {.mantine-visible-from-lg {display: none !important;}}@media (min-width: 75em) {.mantine-hidden-from-lg {display: none !important;}}@media (max-width: 87.99375em) {.mantine-visible-from-xl {display: none !important;}}@media (min-width: 88em) {.mantine-hidden-from-xl {display: none !important;}}
  </style>
  <div
    class="m-4081bf90 mantine-Group-root"
    style="--group-gap: var(--mantine-spacing-md); --group-align: center; --group-justify: flex-start; --group-wrap: wrap;"
  >
    <a
      href="/home"
    >
      <span
        class="mantine-focus-auto m-849cf0da m-b6d8b162 mantine-Text-root mantine-Anchor-root"
        data-underline="hover"
      >
        <p
          class="mantine-focus-auto m-b6d8b162 mantine-Text-root"
        >
          Home
        </p>
      </span>
    </a>
    <a
      href="/meal-plan"
    >
      <span
        class="mantine-focus-auto m-849cf0da m-b6d8b162 mantine-Text-root mantine-Anchor-root"
        data-underline="hover"
      >
        <p
          class="mantine-focus-auto m-b6d8b162 mantine-Text-root"
        >
          Meal plan
        </p>
      </span>
    </a>
    <a
      href="/connections"
    >
      <span
        class="mantine-focus-auto m-849cf0da m-b6d8b162 mantine-Text-root mantine-Anchor-root"
        data-underline="hover"
      >
        <p
          class="mantine-focus-auto m-b6d8b162 mantine-Text-root"
        >
          Connections
        </p>
      </span>
    </a>
  </div>
</div>
`;
